tasks/task_manager.hh

```diff
@@ -70,6 +70,7 @@ public:
     using foreign_task_map = std::unordered_map<task_id, foreign_task_ptr>;
     using module_ptr = shared_ptr<module>;
     using modules = std::unordered_map<std::string, module_ptr>;
+    using virtual_task_hint = std::unordered_map<sstring, sstring>;
```

Comment:
We know exactly what we want to put into the virtual_task_hint. Isn't it better to use the real type

struct virtual_task_hint {
std::optional<table_id> table_id;
std::optional<other_thing_for_the_hint> foo;
}

Reply:
We do know what we want to use for repair tasks but `virtual_task_hint` can be used for all virtual tasks. We could alternatively have hint inheritance and cast the type for each task type, but I like the map more.

Reply:
The problem with the string map is that we have to serialize and deserialize to and from string unnecessarily. 

We can use any tasks we want with the following, without even playing with inheritance and type casks. We can simply add only types that we need for different tasks. Some of fields can be used by different tasks.
```
struct virtual_task_hint {
  std::optional<table_id> table_id;
  std::optional<foo> foo;
  ...
  std::optional<bar> bar;
}

if (!virtual_task_hint.foo) {
     // throw error if foo is expected
}
// use foo after the check
```